### ISSUE TITLE
index out of bounds when delete Chinese characters

### ISSUE THREAD

**Issue by cctse** [curlpipe/ox#71]
**Describe the bug**
ox crash when backspace to delete Chinese characters
```
  tes[+] │
 1  测 thread 'main' panicked at 'index out of bounds: the len is 3 but the index is 4', src/editor.rs:193:50
 ~                           note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
 ~                        zsh: abort      ox tes
 ~                                              %
```

**To Reproduce**
1. sh# ox tes
2. type 测试
3. press backspace

**Expected behavior**
it should delete characters normally

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Mac

**Comment by curlpipe** [comment]
谢谢。I'll look into this immediately.

**Comment by curlpipe** [comment]
I've fixed this in 0.2.7. Thank you.
